components/card/card.css

```diff
@@ -0,0 +1,330 @@
+    align-items: center;
+    justify-content: center;
+    width: max-content;
+    height: max-content;
+    /*position: absolute;*/
+    line-height: 1;
+    font-weight: 500;
+    font-size: 1rem;
+    padding: 0.75rem 0.5rem;
+    margin: 0%;
+    
+}
+
+/*image for cards*/
+.card-img {
+    width: 100%;
+    object-fit: contain;
+}
+
+.img-responsive {
+    max-inline-size: 100%;
+    block-size: auto;
+    font-size: inherit;
+    display: block;
+    max-width: 100%;
+}
+
+
+/* card footer */
+.card-footer {
+    display: flex;
+    flex-direction: row;
+    width: 100%;
+    gap: 0.75rem;
+    flex-wrap: wrap;
+    padding: 0%;
+    margin-bottom: 1rem;
+}
+
+/*simple text card*/
+
+.text-card .card-footer{
+    display: flex;
+    justify-content: space-around;
+    margin: 0%;
+}
+
+.text-card .btn-icon{
+    background: none;
+    padding: 0% 0% 1rem 0%;
+    font-size: 1.5rem;
+    color: var(--default-color);
+}
+
+.text-card .btn-icon:hover{
+    color: var(--primary-color);
+}
+
+
+.card-btn{
+    display: block;
+    z-index: 1;
+    color: var(--default-color);
+    position: absolute;
+    top: 0;
+    right: 0;
+    width: max-content;
+    height: max-content;
+    background-color: transparent;
+    border: none;
+    text-align: center;
+    font-weight: 500;
+    cursor: pointer;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: center;
+    align-items: center;
+    gap: 0.5rem 0.75rem;
+    font-size: 1.8rem;
+    margin-right: 0.5rem;
+    margin-top: 0.8rem;
+}
+
+
+.card-btn:hover{
+    color: var(--primary-color);
+}
+
+
+.btn{
+    margin: 0% 1rem;
+    font-size: 1rem;
+}
+
+.card-footer .btn {
+    height: max-content;
+}
+
+/*card with shadow*/
+.card-shadow {
+    box-shadow: 0 0 0.75rem 0.25rem var(--box-shadow-color);
+}
+
+
+/*card with overlay text*/
+
+.overlay-container {
+    position: relative;
+}
+
+.overlay-text {
+    position: absolute;
+    transform: translate(-50%, -50%);
+    left: 50%;
+    bottom: 50%;
+    top: 50%;
+    right: 50%;
+    z-index: 1;
+    -webkit-transform: translate(-50%, -50%);
+    -moz-transform: translate(-50%, -50%);
+    -ms-transform: translate(-50%, -50%);
+    -o-transform: translate(-50%, -50%);
+    background-color: var(--default-color);
+    color: var(--bg-color);
+    width: max-content;
+    height: max-content;
+    box-shadow: 0px 0px 1rem 0.5rem var(--box-shadow-color);
+    font-size: 1.2rem;
+    font-weight: 600;
+    padding: 0.8rem;
+    border-radius: 0.5rem;
+}
+
+.overlay-container .overlay {
+    display: flex;
+}
+
+.overlay {
+    position: absolute;
+    top: 0;
+    left: 0;
+    right: 0;
+    bottom: 0;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    background-color: var(--overlay-bg);
+    display: none;
+    border-radius: var(--br-5);
+    -webkit-border-radius: var(--br-5);
+    -moz-border-radius: var(--br-5);
+    -ms-border-radius: var(--br-5);
+    -o-border-radius: var(--br-5);
```

Comment:
Had a general query on why are we using these rules for border-radius? Is it because of making it compatible with other browsers?